src/Task/VirtuosoTaskBase.php

```diff
@@ -80,10 +98,64 @@ protected function execute($query) {
   }
 
   /**
-   * Set the permissions of the '/sparql' endpoint to allow update queries.
+   * Returns a GraphStore object of initialized with the object's settings.
+   *
+   * @return \EasyRdf\GraphStore
+   *   The GraphStore object.
    */
-  public function main() {
+  function graphStore() {
+    $connect_string = 'http://' . $this->dsn . ':' . $this->port . '/sparql-graph-crud';
+    // Use a local SPARQL 1.1 Graph Store.
+    return new GraphStore($connect_string);
+  }
+
+  /**
+   * Clears a graph from the endpoint.
+   *
+   * @param string $uri
+   *   A uri representing a graph.
+   *
+   * @throws \EasyRdf\Exception
+   *   Thrown if something went wrong with the delete method.
+   */
+  function deleteGraph($uri) {
+    $this->graphStore()->delete($uri);
+  }
 
+  /**
+   * Replace the contents of a graph in the graph store with new data
+   *
+   * @param string $file_path
+   *   The path of the file that should be parsed and uploaded.
+   * @param string $graph_uri
+   *   The graph uri that the triples should end up to.
+   * @param string $format
+   *   The format of the contents of the file provided by $file_path.
+   */
+  function replaceGraph($file_path, $graph_uri, $format = 'ntriples') {
+    $graph = new Graph();
+    $graph->parseFile($file_path);
+    $this->graphStore()->replace($graph, $graph_uri, $format);
+  }
+
+  /**
+   * Executes a query through an HTTP request.
+   *
+   * @param string $query
+   *   The string version of the query to execute.
+   *
+   */
+  function sparql($query) {
+    // @todo: The port should be passed as a variable below.
+    $connect_string = 'http://' . $this->dsn . ':8890/sparql';
```

Comment:
Isn't this now possible? Or it's a different port from the other one above and below?